docs/design-proposals/cloud-edge-communication/cloud-edge-communication-framework.md

```diff
@@ -30,28 +30,67 @@ edge server -----> Proxy A(in the edge) <----------> Proxy B(in the cloud) -----
+
+We need to add the gateway/proxy for the edge site and the cloud for the communication, we call it edgegateway in each edge site, and in addition to communication, cloudgateway in responsible for central control. We don't expose the network to the cloud or the edge site(cause too much attach surface), we expose the services to the gateway.  
 
-We want to introduce a new concept virtual presence:
+Here we want to introduce a concept "virtual presence":
 * It is a local virtual presentation of remote services
 * It is full isolation of remote servers
-* It exposes services to remote sites, do not expose networks
-* It uses locally assigned virtual presence ips to avoid exposure of site internal addresses
-* It use translation mechanisms to emulate local traffic flow
-* There is no man-in-the-middle, no session termination, it provide end-to-end encryption
 
-![](Proposal-B1.png)
+If the service is exposed to the edge site, there will be a virtual presence for the service in this edge site. The clients in this edge site which want to communicate to the service will be told to communicate to the virtual presence of the service which is on the gateway of the edge site. And our gateway will forward traffic in secured tunnel using local virtual presence ip address to the destination gateway and services.
+
+![](Proposal-B5.png) 
+
+Here is an example for explaining it, in the edge cloud, if we are using openstack to do the vm deploying. In the cloud, there are openstack controller components, for example nova-controller(nova-api/scheduler/conductor), rabbitmq-server, neutron-server and something else. And there are nova-computes running in the edge site. For deploying vm, nova-controller needs to send the vm deploying request to the nova-compute through rabbitmq, and nova-compute need to request to the neutron-server for port creating through rest calls. Only for these two cloud-edge communication: 
+1. There is ip conflict in edge site A and edge site B, the nova-compute server use the same ip address for communication to the cloud.
+2. We can expose the rabbitmq service which is in the cloud to the edge site A and edge site B, it will expose the rabbitmq service url for the edge site. There will be a virtual presence for the rabbitmq service in edge site A, and another virtual presence for the rabbitmq service in edge site B.
+The virtual presence will be assigned a local virtual ip address of the edge site by the controller(CloudGateway), the local virtual ip is in the edge site address spaces which is predefined for the edge site, and it is just used in the edge site, so it is no need to global planning for the cloud.
+3. The nova compute need to use the local dns which is a component of the edgegateway A. Edgegateway A will resolve the rabbitmq service url to the virtual presence ip address which is routed to the edgegateway A. 
```

Comment:
so all nova compute service in all the edge sites will have the same url for rabbitmq. edgegateway will resolve and direct them to their local virtual presence